pinot-broker/src/main/java/org/apache/pinot/broker/api/resources/PinotClientRequest.java

```diff
@@ -141,6 +152,49 @@ public void processSqlQueryPost(String query, @Suspended AsyncResponse asyncResp
     }
   }
 
+  @DELETE
+  @Path("query/{queryId}")
+  @Produces(MediaType.APPLICATION_JSON)
+  @ApiOperation(value = "Cancel a query as identified by the queryId", notes = "No effect if no query exists for the "
+      + "given queryId on the requested broker. Query may continue to run for a short while after calling cancel as "
+      + "it's done in a non-blocking manner. The cancel method can be called multiple times.")
+  @ApiResponses(value = {
+      @ApiResponse(code = 200, message = "Success"), @ApiResponse(code = 500, message = "Internal server error"),
+      @ApiResponse(code = 404, message = "Query not found on the requested broker")
+  })
+  public String cancelQuery(
```

Comment:
just wondering, why you removed the brokerId_ prefix from the id here? Once we make this a controller API, we'll again have to append brokerId to ensure unique ids, and the request from user will also have to include that..

Reply:
The controller would just need to ask a broker to cancel a query by the id the broker returns. 

But yes, user of controller API would need to cancel a query by brokerId and requestId but user can provide both as two params, not need to know how both are stitched together. So that the global unique <brokerId_requestId> is totally internal impl detail.

Reply:
but the brokers could generate same queryId Long. so when controller collects queries from brokers, it'll have to add a prefix anyway before it can present it to the user?

Reply:
Yes, you’re right. I’m thinking the controller API can return a map: brokerId->list of request Ids